pkg/autodiscovery/providers/endpointschecks.go

```diff
@@ -0,0 +1,97 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+// +build kubelet
+
+package providers
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/clusteragent"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// EndpointsChecksConfigProvider implements the ConfigProvider interface
+// for the endpoints check feature.
+type EndpointsChecksConfigProvider struct {
+	dcaClient      *clusteragent.DCAClient
```

Comment:
Why not using the interface here ?

Reply:
that's just because the interface was introduced here https://github.com/DataDog/datadog-agent/pull/3252 after I started this PR. I'll use it here, thanks!

Reply:
done